processor/processortest/nop_processor.go

```diff
@@ -74,8 +74,9 @@ func (npf *NopProcessorFactory) Type() configmodels.Type {
 // CreateDefaultConfig creates the default configuration for the Processor.
 func (npf *NopProcessorFactory) CreateDefaultConfig() configmodels.Processor {
 	return &configmodels.ProcessorSettings{
-		TypeVal: npf.Type(),
-		NameVal: string(npf.Type()),
+		EntityName: configmodels.EntityName{
```

Comment:
This still has a smell to me but this doesn't seem to make it any worse.

Reply:
configmodels.NewEntityName(npf.Type(), "")?